src/components/RoomHeader/index.js

```diff
@@ -2,23 +2,44 @@ import React from 'react'
 import style from './index.module.css'
 
 export const RoomHeader = ({
-  state: { room, user, sidebarOpen, userListOpen },
-  actions: { setSidebar, setUserList },
+  state: { room, user, sidebarOpen, userListOpen, inputIsOpen, roomTitle },
+  actions: { setSidebar, setUserList, setTitleEditable, updateRoom, changeInputTitle },
 }) => (
-  <header className={style.component}>
-    <button onClick={e => setSidebar(!sidebarOpen)}>
-      <svg>
-        <use xlinkHref="index.svg#menu" />
-      </svg>
-    </button>
-    <h1>{room.name && room.name.replace(user.id, '')}</h1>
-    {room.users && (
-      <div onClick={e => setUserList(!userListOpen)}>
-        <span>{room.users.length}</span>
+    <header className={style.component}>
+      <button onClick={e => setSidebar(!sidebarOpen)}>
         <svg>
-          <use xlinkHref="index.svg#members" />
+          <use xlinkHref="index.svg#menu" />
         </svg>
-      </div>
-    )}
-  </header>
-)
+      </button>
+      {
+        (room.name && !inputIsOpen) && <h1 onClick={e => setTitleEditable(!inputIsOpen, room.name)}>
+          {room.name.replace(user.id, '')}
+        </h1>
+      }
+
+      {
+        inputIsOpen &&
+        <h1>{[
```

Comment:
You could un-nest this a bit.. perhaps what we should be aiming for is something like:
```
<form className={style.horizontal}>
  <input />
  <button />
</form>
```

Reply:
@lukejacksonn are you talking about changing the indentation or logic here ?

Reply:
I had wrapped the form in h1 because h1 had styles properties that were used by the heading before editable field is set true. So what I can do now is that to make another style object with same properties and apply it to form. But it will be like duplicating things. Apart from that I don't think that we need a form here either. If I remove it it'll work fine. So, should I remove it ?